### PR TITLE
Fix:  Docstrings, Typehints, os/pathlib, super()

### PR DESCRIPTION
**Pull request by kosonocky**
Helping clean up some things while I read through the codebase.  Here are the types of changes in this PR:

- Removing one instance of super().__init__() that was found in a class that had no inheritance.  From the context, I'm fairly sure this makes sense to remove but wanted to explicitly double check with you all
- Converting os.path to pathlib.Path in setup.py
- Adding typehints when they were absent
- Changing Pass to Return when it made sense to do so (to be consistent w/ other functions)
- Filling in incomplete docstrings

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,19 +21,19 @@
...
         for filename in filenames:
-            paths.append(os.path.join('..', path, filename))
+            paths.append(pathlib.Path('..', path, filename))
     return paths
 
 
 def get_version():
-    import os, re
-    VERSIONFILE = os.path.join(__packagename__, '__init__.py')
+    import os, re   # (Clay) - Why import os and re down here?  Can we just import at top?
```
Comment (michaelosthege) [inline setup.py:35]: Yes, this can be moved.
I've been copying this function across multiple of my projects, that's why I kept the imports inside.

**Review by michaelosthege — COMMENTED** [review]
Thanks!
I never bothered about `-> None` hints, because usually mypy can figure these out itself, but they're legit, of course.

File: setup.py [hunk 2]
```diff
@@ -16,24 +16,24 @@
...
 def package_files(directory):
-    assert os.path.exists(directory)
+    assert pathlib.Path(directory).exists()
     fp_typed = pathlib.Path(__packagename__, 'py.typed')
     fp_typed.touch()
     paths = [str(fp_typed.absolute())]
     for (path, directories, filenames) in os.walk(directory):
         for filename in filenames:
-            paths.append(os.path.join('..', path, filename))
+            paths.append(pathlib.Path('..', path, filename))
```
[the remark this answers is not in the document]
Comment (kosonocky) [reply setup.py:31]: Ah, got it.  Good catch

**Comment by kosonocky** [comment]
> 

Happy to help!  And great, that sounds good.

**Pull request merged**